src/syntax_text.rs

```diff
@@ -0,0 +1,223 @@
+use std::{fmt, ops};
+
+use crate::{
+    cursor::{SyntaxElement, SyntaxNode},
+    SmolStr, TextRange, TextUnit,
+};
+
+#[derive(Clone)]
+pub struct SyntaxText {
```

Comment:
This might need some docs because I have no idea what it is

Reply:
Yeah, the docs are yet nonexistent, will add them.

`SyntaxText` is a "string" type. It represents a text of a single syntax node. It can be useful to do things like "how many newlines are there in this node"?

Reply:
Ah, it recurses children?

Reply:
yep!